distributed/protocol/cupy.py

```diff
@@ -5,6 +5,17 @@
 from .cuda import cuda_serialize, cuda_deserialize
 
 
+class PatchedCudaArrayInterface(object):
+    # TODO: This class wont be necessary
+    #       once Cupy<7.0 is no longer supported
+    def __init__(self, ary):
+        cai = ary.__cuda_array_interface__
+        cai_cupy_vsn = cupy.ndarray(None).__cuda_array_interface__["version"]
```

Comment:
This is a good idea, but `cupy.ndarray(0)` is a better choice as it doesn't incur a memory allocation, whereas `cupy.ndarray(None)` does. The pointer for the first is a null pointer, the latter has a valid one, as seen below (and verifiable with nvprof):

```python
In [1]: import cupy

In [2]: cupy.ndarray(0).__cuda_array_interface__["data"]
Out[2]: (0, False)

In [3]: cupy.ndarray(None).__cuda_array_interface__["data"]
Out[3]: (140350232461312, False)
```

Reply:
Thanks for pointing this out @pentschev !!   My mistake :)

Reply:
No worries, it's something I didn't know as well, but I decided to look if that would incur in any unwanted overhead. Thanks for the fix!